### PR TITLE
EOS-19179: Update the manpage for m0d with all the supported options.

### PR DESCRIPTION
**Pull request by mehjoshi**
Updated the manpage of m0d with all the supported options.

Signed-off-by: Mehul Joshi <mehul.joshi@seagate.com>

### CODE REVIEW

**Comment by madhavemuri** [comment]
@mehjoshi: Can you fix man of motr utility as well,

{noformat}
TR(7)                                                             Motr User Manual                                                             MOTR(7)

NAME
       motr - Motr object stor

DESCRIPTION
       Motr object stor [...to be filled]

   Motr utilities
       m0d
           Motr file system service daemon.

       m0console
           Blah blah...

       m0addb2dump
           Blah blah...

       m0tracedump
           Utility to convert binary trace logs to YAML format for further analysis by m0trace.

       m0traced
           Daemon, which polls trace buffer of m0tr.ko or m0d and saves trace records in binary format into continuous trace file for postmortem
           analysis by m0tracedump and m0trace utilities.

       m0trace
           Utility to filter, analyze and print in human readable form trace logs in YAML format.

AUTHORS
       See the Motr distribution AUTHORS file.

BUGS
       All bugs are added by Motr Team ;)

RESOURCES
       Main web site: http://xyratex.com/

COPYING

{noformat}

**Comment by mehjoshi** [comment]
> @mehjoshi: Can you fix man of motr utility as well,
> 
> {noformat}
> TR(7) Motr User Manual MOTR(7)
> 
> NAME
> motr - Motr object stor
> 
> DESCRIPTION
> Motr object stor [...to be filled]
> 
> Motr utilities
> m0d
> Motr file system service daemon.
> 
> ```
>    m0console
>        Blah blah...
> 
>    m0addb2dump
>        Blah blah...
> 
>    m0tracedump
>        Utility to convert binary trace logs to YAML format for further analysis by m0trace.
> 
>    m0traced
>        Daemon, which polls trace buffer of m0tr.ko or m0d and saves trace records in binary format into continuous trace file for postmortem
>        analysis by m0tracedump and m0trace utilities.
> 
>    m0trace
>        Utility to filter, analyze and print in human readable form trace logs in YAML format.
> ```
> 
> AUTHORS
> See the Motr distribution AUTHORS file.
> 
> BUGS
> All bugs are added by Motr Team ;)
> 
> RESOURCES
> Main web site: http://xyratex.com/
> 
> COPYING
> 
> {noformat}

Done.

**Review by madhavemuri — APPROVED** [review]

File: doc/man/motr.7.txt [hunk 1]
```diff
@@ -10,19 +10,23 @@ motr - Motr object stor
...
+This manpage briefly describes the major binaries and utilities available in
+Motr object stor.
 
 Motr utilities
 ~~~~~~~~~~~~~~
 
 *m0d*::
-  Motr file system service daemon.
+  Motr file system service daemon. This command initializes and starts
+  requested Motr file system service as a daemon process in user space.
```
Comment (huanghua78) [inline doc/man/motr.7.txt:21]:
```
Motr service daemon. This command initializes and starts requested
Motr service as a daemon process in user space.
```
Remove the "file system". It is not a file system any longer, or it is misleading.

File: doc/man/m0d.1.txt [hunk 2]
```diff
@@ -21,24 +21,172 @@ service as a daemon process in user space.
...
 *-l*::
     List supported services.
 
-*-Q*::
-    Minimum TM Receive queue length.
+*-v*::
+    Show version information and exit.
 
 *-h, --help*::
     Display help screen and exit.
```
Comment (huanghua78) [inline doc/man/m0d.1.txt:37]: Please sort these in alphabet order.

File: doc/man/m0d.1.txt [hunk 3]
```diff
@@ -21,24 +21,172 @@ service as a daemon process in user space.
...
+    Force mkfs to override found filesystem.
+
+*-Q* num::
+    Minimum length of TM receive queue.
+
+*-M* num::
+    Maximum RPC message size.
+
+*-H* addr::
+    Endpoint address of HA service.
```
Comment (huanghua78) [inline doc/man/m0d.1.txt:51]: Please sort these in alphabet order.

Or the man page will sort them automatically?

Comment (mehjoshi) [reply doc/man/m0d.1.txt]: Done. Sorted and updated.

File: doc/man/motr.7.txt (hunk 1 above)
Comment (mehjoshi) [reply doc/man/motr.7.txt]: Done.

**Review by huanghua78 — APPROVED** [review]

**Comment by gowthamchinna** [comment]
test this please

**Comment by mehjoshi** [comment]
> test this please

What test are you suggesting? I have built and tested the changes in the dev environment before committing every patch.

**Comment by gowthamchinna** [comment]
> > test this please
> 
> What test are you suggesting? I have built and tested the changes in the dev environment before committing every patch.

@mehjoshi this is trigger phrase for automated PR 1N deployment in jenkins. you can skip considering such phrase from my ID

**Pull request merged**